app/auth/callback/route.ts

```diff
@@ -6,8 +6,9 @@ import * as Sentry from "@sentry/nextjs";
 import { TOAST_REDIRECT_KEY } from "@/constants";
 
 export async function GET(request: NextRequest) {
-  const { searchParams } = new URL(request.url);
+  const { searchParams, origin } = new URL(request.url);
   const token_hash = searchParams.get("token_hash");
+  const code = searchParams.get("code");
```

Comment:
in the interest of my head not hurting, let's put OAuth handing in `app/oauth/callback/route.ts` and magic link in `app/magiclink/callback/route.ts`